src/EditorFeatures/Core/InlineRename/CommandHandlers/AbstractRenameCommandHandler_RenameHandler.cs

```diff
@@ -92,14 +105,41 @@ private void ExecuteRenameWorker(RenameCommandArgs args, CommandExecutionContext
             // There can be zero selectedSpans in projection scenarios.
             if (selectedSpans.Count != 1)
             {
-                ShowErrorDialog(workspace, EditorFeaturesResources.You_must_rename_an_identifier);
+                await ShowErrorDialogAsync(workspace, EditorFeaturesResources.You_must_rename_an_identifier).ConfigureAwait(false);
                 return;
             }
 
-            var sessionInfo = _renameService.StartInlineSession(document, selectedSpans.Single().Span.ToTextSpan(), cancellationToken);
+            var sessionInfo = await _renameService.StartInlineSessionAsync(document, selectedSpans.Single().Span.ToTextSpan(), cancellationToken).ConfigureAwait(false);
             if (!sessionInfo.CanRename)
             {
-                ShowErrorDialog(workspace, sessionInfo.LocalizedErrorMessage);
+                await ShowErrorDialogAsync(workspace, sessionInfo.LocalizedErrorMessage).ConfigureAwait(false);
+                return;
+            }
+
+            return;
+
+            //
+            // Local Functions
+            //
+            IUIThreadOperationContext GetContext()
+            {
+                var backgroundWorkIndicatorFactory = workspace.Services.GetService<IBackgroundWorkIndicatorFactory>();
+
+                // Rename command handler builds for hosts that don't necessarily
+                // have an implementation for IBackgroundWorkIndicatorFactory
+                if (backgroundWorkIndicatorFactory is null)
```

Comment:
When will this be null?

Reply:
actually, this is true.  it's never null.  in the case where we don't have our own indicator, we just create a TWD and wrap that :D

Reply:
Do we??? I didn't know... I can change then :)